src/platform/nrfconnect/OTAImageProcessorImpl.cpp

```diff
@@ -45,23 +46,12 @@ CHIP_ERROR OTAImageProcessorImpl::PrepareDownloadImpl()
     ReturnErrorOnFailure(System::MapErrorZephyr(dfu_target_mcuboot_set_buf(mBuffer, sizeof(mBuffer))));
     ReturnErrorOnFailure(System::MapErrorZephyr(dfu_target_reset()));
 
-    return System::MapErrorZephyr(dfu_target_init(DFU_TARGET_IMAGE_TYPE_MCUBOOT, /* size */ 0, nullptr));
+    // Initialize dfu target to receive first image
+    return System::MapErrorZephyr(dfu_target_init(DFU_TARGET_IMAGE_TYPE_MCUBOOT, mCurrentImage, /* size */ 0, nullptr));
 }
 
 CHIP_ERROR OTAImageProcessorImpl::Finalize()
 {
-    return CHIP_NO_ERROR;
-}
-
-CHIP_ERROR OTAImageProcessorImpl::Abort()
-{
-    return System::MapErrorZephyr(dfu_target_reset());
-}
-
-CHIP_ERROR OTAImageProcessorImpl::Apply()
-{
-    ReturnErrorOnFailure(System::MapErrorZephyr(dfu_target_done(true)));
-
```

Comment:
Why was this changed? The reboot should happen only after `Apply`, not after finalizing the download.